ibis/pyspark/client.py

```diff
@@ -19,30 +21,39 @@ def __init__(self, session):
         super().__init__(session)
         self.translator = PySparkExprTranslator()
 
-    def compile(self, expr, params=None, *args, **kwargs):
+    def compile(self, expr, timecontext=None, params=None, *args, **kwargs):
         """Compile an ibis expression to a PySpark DataFrame object
         """
 
         # Insert params in scope
         if params is None:
-            scope = {}
+            scope = Scope()
         else:
-            scope = dict(
-                (param.op(), raw_value) for param, raw_value in params.items()
+            scope = Scope().merge_scopes(
+                [
+                    make_scope(param.op(), raw_value, None)
```

Comment:
I think `make_scope` should take "op, timecontext, value" to be more consistent with other methods like `set_value`

Reply:
Make sense, just a note that conceptually speaking, `value` is determined by `op` and `timecontext`, so we consider `op` and `timecontext` to be "key"s and `value` to be value.  So it is fine to do so in API exposed in Scope class.

In our implementation, `op` is the only key. User shouldn't care about this and they should treat op and timecontext as keys when using this class.